source/common/network/address_impl.cc

```diff
@@ -117,6 +119,10 @@ Ipv4Instance::Ipv4Instance(const std::string& address, const SocketInterface* so
 Ipv4Instance::Ipv4Instance(const std::string& address, uint32_t port,
                            const SocketInterface* sock_interface)
     : InstanceBase(Type::Ip, sockInterfaceOrDefault(sock_interface)) {
+  absl::Status status = validateProtocolSupported();
+  if (!status.ok()) {
```

Comment:
factor out throwOnError(absl::Status) and use it several times rather than repeating this multi-line pattern.

Put `ASSERT(!isMainThread...)` into that helper.